tf2_ros/src/static_transform_broadcaster_program.cpp

```diff
@@ -35,68 +35,172 @@
+    "  --yaw YAW             yaw component of an Euler representation of the rotation (RPY)";
+  printf("%s\n", usage);
+}
+
+tf2::Quaternion _get_rotation(const std::unordered_map<std::string, std::string> & args)
+{
+  tf2::Quaternion quat;
+  bool found_quaternion = false;
+  bool found_rpy = false;
+  auto iter = args.find("--qx");
+  if (iter != args.end()) {
+    quat.setX(std::stod(iter->second));
+    found_quaternion = true;
+  }
+  iter = args.find("--qy");
+  if (iter != args.end()) {
+    quat.setY(std::stod(iter->second));
+    found_quaternion = true;
+  }
+  iter = args.find("--qz");
+  if (iter != args.end()) {
+    quat.setZ(std::stod(iter->second));
+    found_quaternion = true;
+  }
+  iter = args.find("--qw");
+  if (iter != args.end()) {
+    quat.setW(std::stod(iter->second));
+    found_quaternion = true;
+  }
+  /* otherwise, look for roll, pitch, yaw */
+  double roll = 0.0;
+  double pitch = 0.0;
+  double yaw = 0.0;
+  iter = args.find("--roll");
+  if (iter != args.end()) {
+    roll = std::stod(iter->second);
+    found_rpy = true;
+  }
+  iter = args.find("--pitch");
+  if (iter != args.end()) {
+    pitch = std::stod(iter->second);
+    found_rpy = true;
+  }
+  iter = args.find("--yaw");
+  if (iter != args.end()) {
+    yaw = std::stod(iter->second);
+    found_rpy = true;
+  }
+  if (found_quaternion && found_rpy) {
+    RCUTILS_LOG_ERROR("cannot mix euler and quaternion arguments");
+    _print_usage();
+    exit(1);
+  } else if (!found_quaternion) {
+    quat.setRPY(roll, pitch, yaw);
+  }
+  return quat;
+}
+
+tf2::Vector3 _get_translation(const std::unordered_map<std::string, std::string> & args)
+{
+  tf2::Vector3 trans;
+  auto iter = args.find("--x");
+  if (iter != args.end()) {
+    trans.setX(std::stod(iter->second));
+  }
+  iter = args.find("--y");
+  if (iter != args.end()) {
+    trans.setY(std::stod(iter->second));
+  }
+  iter = args.find("--z");
+  if (iter != args.end()) {
+    trans.setZ(std::stod(iter->second));
+  }
+  return trans;
+}
+
 int main(int argc, char ** argv)
 {
   // Initialize ROS
   std::vector<std::string> args = rclcpp::init_and_remove_ros_arguments(argc, argv);
+  std::unordered_map<std::string, std::string> arg_map = _make_arg_map(std::move(args));
   rclcpp::NodeOptions options;
   std::shared_ptr<tf2_ros::StaticTransformBroadcasterNode> node;
-
-  if (args.size() != 9 && args.size() != 10) {
-    printf("A command line utility for manually sending a transform.\n");
-    printf("Usage: static_transform_publisher x y z qx qy qz qw frame_id child_frame_id \n");
-    printf("OR \n");
-    printf("Usage: static_transform_publisher x y z yaw pitch roll frame_id child_frame_id \n");
-    RCUTILS_LOG_ERROR(
-      "static_transform_publisher exited due to not having the right number of arguments");
-    return 2;
+  tf2::Quaternion rotation;
+  tf2::Vector3 translation;
+  try {
+    rotation = _get_rotation(arg_map);
+    translation = _get_translation(arg_map);
```

Comment:
What happens if the user supplies arguments that we don't understand, like `--foo`?

Reply:
I think this is actually immune to that. It should set to the identity transform.

That said, it should probably error, since that's not a valid argument.